docs/RFD-JobWorkerService.md

```diff
@@ -0,0 +1,366 @@
+    - `Memory` - set by writing the provided IOBytesPerSecond to `/sys/fs/cgroup/<uuid>/tasks/memory.max`
+    - `IO` - is set by writing IOBytesPerSecond into `<MAJOR NUMBER OF PHYSICAL DEVICE>:<MINOR NUMBER OF PHYSICAL DEVICE> rbps=<IOBytesPerSecond> wbps=<IOBytesPerSecond> riops=max wiops=max` to `/sys/fs/cgroup/<UUID>/tasks/io.max` <br/><br/> _Note_:  `<MAJOR NUMBER OF PHYSICAL DEVICE>` and `<MINOR NUMBER OF PHYSICAL DEVICE>` shoudl be dynamically detected and provided during runtime.
+  - configures a new [exec.Cmd](https://pkg.go.dev/os/exec#Cmd)
+    - sets clone flags for creating new mount, pid, and network namespaces 
+    - sets [unshare flags](https://pkg.go.dev/syscall#SysProcAttr) for mount so that new mounts are not reflected in host mount namespace 
+    - sets [UseCgroupFD and CgroupFD](https://pkg.go.dev/syscall#SysProcAttr) to the cgroup's file descriptor such as `/sys/fs/cgroup/<UUID>/tasks`
+  - runs the configured `exec.Cmd` in a new goroutine
+    - does not wait for the command to complete 
+    - the goroutine will stop when the process completes or stopped
+
+
+* **func (\*Job) Status() JobStatus** - returns the current status of the job (see [JobStatus](#type-JobStatus))
+
+
+* **func (\*Job) Stream() OutputReader** - returns an `OutputReader` (type that implements [io.Reader](https://pkg.go.dev/io#Reader)). 
+`OutputReader` return entire output from when job started to the current output. `Stream` can call when job started, executing or completed.
+Calling `Stream` for not-existing or not-started job returns empty `OutputReader`.
+
+
+* **func (\*Job) Stop() error** - attempt to gracefully shutdown the process.
+`Stop` will do nothing if the process has already completed or been killed.
+
+### API 
+The API is nothing more than just a wrapper around Library, but main API responsibility provide mTLS Authorization and using Library
+functionality via [gRPC](https://grpc.io/)
+
+#### Following functions considered to be used for API Server.
+
+* `Start` New Job
+  * verify user credentials via mTLS Authorization
+  * create new instance of the `Job` and assign new UUID 
+  * store `job` in sync.Map using UUID as key
+  * start command execution
+  * return UUID to the client
+* `Status` of the Job
+  * return current status of the `Job`
+* `Stream` latest Job output
+  * lookup `Job` by provided UUID
+  * request instance of `OutputReader` using library
+  * send steam output to the client/cli
+* `Stop` Job
+  * lookup `Job` by provided UUID
+  * call `Stop` function if `Job` was found, otherwise return error
+  
+  > **Not any job and it output removed from then server**. So, depend on the load and size of total output it would come to OOM situation sooner or later and [panic](https://go.dev/blog/defer-panic-and-recover).
+  > Probably we need to consider some API or process to clean completed jobs based on size, date of creation, frequency of usage or etc.
```

Comment:
Great that you brought this up. For the purpose of this change, you don’t need to worry about cleanup so for simplify please keep this out of scope.

Reply:
@smallinsky  marked this part as "out of scope".